packages/grainIntegration-merkle/contracts/MerkleRedeem.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity 0.6.8;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract MerkleRedeem is AccessControl {
+    bytes32 public constant DELAYER_ROLE = keccak256("DELAYER_ROLE");
+    bytes32 public constant SEEDER_ROLE = keccak256("SEEDER_ROLE");
+
+    IERC20 public token;
+
+    event Claimed(address _claimant, uint256 _balance);
+
+    // Recorded weeks
+    mapping(uint => bytes32) public weekMerkleRoots;
```

Comment:
It seems like there isn't tight coupling between actual calendar weeks and these indexes. Could we generalize the "week" variable names to something more fitting like "distribution" or "seed" or "allocation"